### PR TITLE
Upgrade to 1.18.2

### LINKED ISSUE #4: Update to 1.18

### PR DESCRIPTION
**Pull request by braxtonhall**
resolves #4 

Hello @Yunus1903 !

This branch successfully builds and has been tested on Minecraft 1.18.2.
I created this PR more as a proof of concept, as I have never written a Minecraft mod and really don't understand the ecosystem at all. There's a little bit of jank in here.

Please adjust the base branch when you create a 1.18 one

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -1,46 +1,53 @@
-import se.bjurr.gitchangelog.plugin.gradle.GitChangelogTask
```
Comment (braxtonhall) [inline build.gradle:1]: This file was recreated by copying the base forge `build.gradle` and reintroducing function calls until things worked again. I assume I missed some important steps

File: gradlew [hunk 2]
```diff
@@ -1,78 +1,129 @@
-#!/usr/bin/env sh
```
Comment (braxtonhall) [inline gradlew:1]: Gradle itself needed to be updated

File: src/main/java/com/yunus1903/chatembeds/client/ChatScrollPos.java [hunk 3]
```diff
@@ -0,0 +1,13 @@
+package com.yunus1903.chatembeds.client;
+
+public class ChatScrollPos {
```
Comment (braxtonhall) [inline src/main/java/com/yunus1903/chatembeds/client/ChatScrollPos.java:3]: This class will probably need to go.

I don't know how to access a `private` field in a Minecraft class, so this is used to get around `chatScrollbarPos` being private in `ChatComponent`

File: src/main/java/com/yunus1903/chatembeds/client/embed/AnimatedImageEmbed.java [hunk 4]
```diff
@@ -70,26 +73,26 @@ List<? extends ChatLine<IReorderingProcessor>> createChatLines()
...
                 private int time;
                 private int lastFrameIndex;
                 private int lastScrollPos;
 
                 @Override
-                public void render(Minecraft mc, MatrixStack matrixStack, int x, int y)
+                public void render(Minecraft mc, PoseStack matrixStack, int x, int y)
                 {
-                    if (lastScrollPos != mc.ingameGUI.getChatGUI().scrollPos)
+                    if (lastScrollPos != getScrollPos())
```
Comment (braxtonhall) [inline src/main/java/com/yunus1903/chatembeds/client/embed/AnimatedImageEmbed.java:85]: This is the same hack mentioned above, `chatScrollbarPos` is private

File: src/main/java/com/yunus1903/chatembeds/client/mixin/NewChatGuiMixin.java [hunk 5]
```diff
@@ -217,61 +214,71 @@ public void func_238493_a_(ITextComponent chatComponent, int chatLineId, int tic
...
-    public void refreshChat(CallbackInfo ci)
+    @Inject(method = "rescaleChat", at = @At("HEAD"))
+    public void rescaleChat(CallbackInfo ci)
     {
         index = 0;
     }
 
+    @Inject(method = "resetChatScroll", at = @At("RETURN"))
+    public void resetChatScroll(CallbackInfo ci) {
+        setScrollPos(chatScrollbarPos);
```
Comment (braxtonhall) [inline src/main/java/com/yunus1903/chatembeds/client/mixin/NewChatGuiMixin.java:269]: Same hack mentioned above to expose `chatScrollbarPos`

File: src/main/java/com/yunus1903/chatembeds/client/screen/AnimatedImageEmbedScreen.java [hunk 6]
```diff
@@ -41,9 +42,10 @@ public void render(MatrixStack matrixStack, int mouseX, int mouseY, float partia
             else currentFrame++;
         }
 
-        minecraft.getTextureManager().bindTexture(embed.getFrames().get(currentFrame).getResourceLocation());
-        AbstractGui.blit(matrixStack, (width - scaledImageWidth) / 2,
+        RenderSystem.setShaderTexture(0, embed.getFrames().get(currentFrame).getResourceLocation());
+        RenderSystem.enableBlend();
```
Comment (braxtonhall) [inline src/main/java/com/yunus1903/chatembeds/client/screen/AnimatedImageEmbedScreen.java:46]: _to be honest_ I'm not even sure how this worked before. It really seems like this should have been required the whole time unless you were taking advantage of some temporal coupling.

**Review by braxtonhall — COMMENTED** [review]
In this review is a little bit of commentary for the changes I made so far

**Comment by Yunus1903** [comment]
Hi,

Cool to see you're interested and devoted to update the mod.
I wanted to do it myself as well but sadly haven't had time or motivation to do so.

I will try to review, cleanup and work on it as well in the next few days. Since I won't be having much time this weekend, most of it might happen next week.
I'll try to give as much feedback on here as possible and thanks for putting in time and effort to work on the mod 😅.

File: src/main/java/com/yunus1903/chatembeds/client/ChatScrollPos.java (hunk 3 above)
Comment (Yunus1903) [reply src/main/java/com/yunus1903/chatembeds/client/ChatScrollPos.java:3]: You can transform `private` fields to be more permissive using [Access Transformers](https://mcforge.readthedocs.io/en/latest/advanced/accesstransformers/).

**Comment by braxtonhall** [comment]
closing to get this out of my pull requests tab (I assume this will not get looked at again)

**Pull request closed without merging**